src/components/baseComponent.ts

```diff
@@ -141,4 +184,140 @@ export abstract class MgtBaseComponent extends LitElement {
     });
     this.dispatchEvent(event);
   }
+
+  /**
+   * Reload the state.
+   * Returns false if already loading, unless forced.
+   * Use reload instead of load to ensure loading events are fired.
+   *
+   * @protected
+   * @returns {Promise<void>}
+   * @memberof MgtBaseComponent
+   */
+  protected async reloadState(force: boolean = false): Promise<boolean> {
+    if (!force && this._isLoadingState) {
```

Comment:
I'm not sure about the `force` parameter here. In general, If reloadState is called, I've called it because something changed and I need to reload and I want to make sure it will happen. Returning `false` here is not helping as it's pushing that work on me to figure that out.

I'm thinking about this similar to how `render` works. Whenever a request to update happens while the component is rendering, that request is flagged and the render method runs again when possible. The current render task is not canceled.

To follow the same pattern, instead of calling this `reloadState`, what if we call it `requestStateUpdate`? Passing force will try to cancel the current running `loadState` - just like how it is now. But not passing anything (or `force=false`) would let the current running `loadState` to finish and then try to `loadState` again.

Reply:
That's a good point, and I like the idea for `requestStateUpdate`! I've made a few changes to include this consideration.